File: Tema 05 - Typescript/PetShelterBackend/PetShelter.Domain/Donation.cs
namespace PetShelter.Domain
{
    public class Donation
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Amount { get; set; }
        public int DonorId { get; set; }
        public int PetId { get; set; }

        public Person Donor { get; set; }
        public Fund Fund { get; set; }

        public Donation(decimal amount, int id = 0)
        {
            Amount = amount;
            Id = id;
        }
    }
}

File: Tema 06 - React/petshelter.ui/src/Models/Donation.ts
import type { Fund } from "./Fund";
import type { Person } from "./Person";

export class Donation {
    name: string;
    amount: number;
    donor?: Person;
    fund?: Fund;

    constructor(name: string, amount: number, donor?: Person, fund?: Fund) {
        this.name = name;
        this.amount = amount;
        this.donor = donor;
        this.fund = fund;
    }
}